lib/rack/linkeddata/conneg.rb

```diff
@@ -124,6 +124,8 @@ def find_writer_for_content_type(content_type)
           RDF::Writer.for(:content_type => (content_type = options[:default]))
         when 'text/*'
           RDF::Writer.for(:content_type => (content_type = 'text/turtle'))
+        when 'application/*'
+          RDF::Writer.for(:content_type => (content_type = 'application/ld+json'))
         when /^([^\/]+)\/\*$/
```

Comment:
OTOH, it is unlikely that there will be formats specifically responding to application/xml or application/json, so probably no need for the check, so it could be folded into the existing when clauses.